src/fsharp/CompileOps.fs

```diff
@@ -1639,15 +1639,111 @@ let GetDefaultFSharpCoreReference() = typeof<list<int>>.Assembly.Location
 
 // If necessary assume a reference to the latest System.ValueTuple with which those tools are built.
 let GetDefaultSystemValueTupleReference() = 
-    try 
-       let asm = typeof<System.ValueTuple<int,int>>.Assembly
-       if asm.FullName.StartsWith "System.ValueTuple" then 
-           Some asm.Location 
-       else None
-    with _ -> None
+    let path = Path.GetDirectoryName(GetDefaultFSharpCoreReference())
```

Comment:
And it won't work on the dotnet cli.  Since we don't co-locate system.valuetuple.dll